TAF/utils/scripts/docker/get-compose-file.sh

```diff
@@ -7,39 +7,37 @@ USE_SECURITY=${3:--}
 
 # # nightly or other release
 mkdir temp
 if [ "$USE_RELEASE" = "nightly-build" ]; then
-     COMPOSE_FILE="docker-compose-nexus${USE_DB}${USE_NO_SECURITY}${USE_ARM64}.yml"
-     wget -O ${COMPOSE_FILE} "https://raw.githubusercontent.com/edgexfoundry/developer-scripts/master/releases/nightly-build/compose-files/${COMPOSE_FILE}"
+  # generate single file docker-compose.yml for target configuration without
+  # default device services, i.e. no device-virtual service
+  ./sync-nightly-build.sh no-ds ${ARM64_OPTION} ${NO_SECTY_OPTION}
+  mv docker-compose.yml temp/docker-compose-temp.yaml
 
-     cp ${COMPOSE_FILE} temp/docker-compose-temp.yaml
-     # Delete device-virtual service from the compose file
-     sed -i -r '/device-virtual:/,/- command/ {/ / d;}' temp/docker-compose-temp.yaml
+  # Insert device services into the compose file
+  sed -e '/app-service-rules:/r docker-compose-device-service.yaml' -e //N temp/docker-compose-temp.yaml > temp/device-service-temp.yaml
 
-    # Insert device services into the compose file
-    sed -e '/# device-random:/r docker-compose-device-service.yaml' -e //N temp/docker-compose-temp.yaml > temp/device-service-temp.yaml
-
-    # Insert required services for end to end tests
-    sed -e '/# device-random:/r docker-compose-end-to-end.yaml' -e //N temp/device-service-temp.yaml > docker-compose.yaml
+  # Insert required services for end to end tests
+  sed -e '/app-service-rules:/r docker-compose-end-to-end.yaml' -e //N temp/device-service-temp.yaml > docker-compose.yaml
 
 else
-     COMPOSE_FILE="docker-compose-${USE_RELEASE}${USE_DB}${USE_NO_SECURITY}${USE_ARM64}.yml"
-     wget -O ${COMPOSE_FILE} "https://raw.githubusercontent.com/edgexfoundry/developer-scripts/master/releases/${USE_RELEASE}/compose-files/${COMPOSE_FILE}"
+  COMPOSE_FILE="docker-compose-${USE_RELEASE}${USE_DB}${USE_NO_SECURITY}${USE_ARM64}.yml"
+  wget -O ${COMPOSE_FILE} "https://raw.githubusercontent.com/edgexfoundry/developer-scripts/master/releases/${USE_RELEASE}/compose-files/${COMPOSE_FILE}"
 
-     cp ${COMPOSE_FILE} temp/docker-compose-temp.yaml
+  cp ${COMPOSE_FILE} temp/docker-compose-temp.yaml
 
-     sed -n '/Service_Host: edgex-device-virtual/,/edgex-device-modbus:/{//!p;}' docker-compose-device-service.yaml > temp/device-virtual.yaml
+  sed -n '/Service_Host: edgex-device-virtual/,/edgex-device-modbus:/{//!p;}' docker-compose-device-service.yaml > temp/device-virtual.yaml
 
-     sed -n \
-         -e '1,/Service_Host: edgex-device-virtual/ p' \
-         -e '/Service_Host: edgex-device-virtual/ r temp/device-virtual.yaml' \
-         -e '/#  device-random:/,$ p' \
-         temp/docker-compose-temp.yaml > docker-compose.yaml
+  sed -n \
+     -e '1,/Service_Host: edgex-device-virtual/ p' \
+     -e '/Service_Host: edgex-device-virtual/ r temp/device-virtual.yaml' \
+     -e '/#  device-random:/,$ p' \
+     temp/docker-compose-temp.yaml > docker-compose.yaml
 fi
```

Comment:
@lenny-intel When generating compose file by geneva version, also need to update `redis` to `database`.

Reply:
Ahh, yes. Also need to handle `mongo` if that is selected... Done. THX again for help with validating this PR!!!